src/main/java/bisq/core/payment/validation/AltCoinAddressValidator.java

```diff
@@ -19,60 +19,37 @@
-import bisq.core.payment.validation.altcoins.PNCAddressValidator;
-import bisq.core.payment.validation.altcoins.WMCCAddressValidator;
-import bisq.core.payment.validation.altcoins.XCNAddressValidator;
-import bisq.core.payment.validation.altcoins.YTNAddressValidator;
-import bisq.core.payment.validation.params.ACHParams;
-import bisq.core.payment.validation.params.AlcParams;
-import bisq.core.payment.validation.params.CageParams;
-import bisq.core.payment.validation.params.CreaParams;
-import bisq.core.payment.validation.params.ICHParams;
-import bisq.core.payment.validation.params.IOPParams;
-import bisq.core.payment.validation.params.ODNParams;
-import bisq.core.payment.validation.params.OctocoinParams;
-import bisq.core.payment.validation.params.OnionParams;
-import bisq.core.payment.validation.params.PARTParams;
-import bisq.core.payment.validation.params.PNCParams;
-import bisq.core.payment.validation.params.PhoreParams;
-import bisq.core.payment.validation.params.PivxParams;
-import bisq.core.payment.validation.params.SpeedCashParams;
-import bisq.core.payment.validation.params.StrayaParams;
-import bisq.core.payment.validation.params.TerracoinParams;
-import bisq.core.payment.validation.params.WACoinsParams;
-import bisq.core.payment.validation.params.WMCCParams;
-import bisq.core.payment.validation.params.XspecParams;
+import bisq.core.payment.validation.altcoins.*;
+import bisq.core.payment.validation.params.*;
 import bisq.core.payment.validation.params.btc.BTGParams;
 import bisq.core.payment.validation.params.btc.BtcMainNetParamsForValidation;
 import bisq.core.util.validation.InputValidator;
-
-import org.libdohj.params.DashMainNetParams;
-import org.libdohj.params.DashRegTestParams;
-import org.libdohj.params.DashTestNet3Params;
-import org.libdohj.params.DogecoinMainNetParams;
-import org.libdohj.params.LitecoinMainNetParams;
-import org.libdohj.params.LitecoinRegTestParams;
-import org.libdohj.params.LitecoinTestNet3Params;
-
+import lombok.extern.slf4j.Slf4j;
 import org.bitcoinj.core.Address;
 import org.bitcoinj.core.AddressFormatException;
 import org.bitcoinj.core.Base58;
 import org.bitcoinj.params.MainNetParams;
 import org.bitcoinj.params.RegTestParams;
 import org.bitcoinj.params.TestNet3Params;
-
-import lombok.extern.slf4j.Slf4j;
-
 import org.jetbrains.annotations.NotNull;
+import org.libdohj.params.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
 
 @Slf4j
 public final class AltCoinAddressValidator extends InputValidator {
 
     private String currencyCode;
+    private final Map<String, SpecificAltCoinAddressValidator> validators = new HashMap<>();
+
+    public AltCoinAddressValidator() {
+        final ServiceLoader<SpecificAltCoinAddressValidator> loader = ServiceLoader.load(SpecificAltCoinAddressValidator.class);
+        for (SpecificAltCoinAddressValidator validator : loader) {
+            validators.put(validator.getCurrencyCode(), validator);
+        }
```

Comment:
A `List<SpecificAltCoinAddressValidator>` should be constructor-injected here.

Reply:
(as opposed to doing the `ServiceLoader` lookup directly within the ctor)

Reply:
You mean that there should be something else registered with Guice that would provide a list of validators?

Reply:
No, I just mean normal injection, i.e. classes should take their dependencies through constructors, not create them themselves. _How_ those deps get wired up is a different matter. Guice might have a role to play there, maybe not. I'd have to look / think about it.

Reply:
Should we call it `AssetProviderRegistry`?